roottest-cling-parsing-semicolon  is not running, because semicolon is special in cmake?

### Check duplicate issues.

- [x] Checked for duplicates

### Description

Trying to debug this test failure [roottest-cling-parsing-semicolon](https://lcgapp-services.cern.ch/cdash/testSummary.php?project=3&name=roottest-cling-parsing-semicolon&date=2025-12-16)

```
-- TEST COMMAND -- 
cd /build/jenkins/workspace/lcg_nightly_pipeline/build/projects/ROOT-HEAD/src/ROOT-HEAD-build/roottest/cling/parsing
/usr/bin/timeout -s USR2 270s /build/jenkins/workspace/lcg_nightly_pipeline/build/projects/ROOT-HEAD/src/ROOT-HEAD-build/bin/root.exe -e '#define ClingWorkAroundMissingDynamicScope' -e '#define ClingWorkAroundUnnamedInclude' -e '#define ClingWorkAroundMissingSmartInclude' -e '#define ClingWorkAroundNoDotInclude' -e '#define ClingWorkAroundMissingAutoLoadingForTemplates' -e '#define ClingWorkAroundAutoParseUsingNamespace' -e '#define ClingWorkAroundTClassUpdateDouble32' -e '#define ClingWorkAroundAutoParseDeclaration' -e '#define ClingWorkAroundMissingUnloading' -e '#define ClingWorkAroundBrokenUnnamedReturn' -e '#define ClingWorkAroundUnnamedDetection2' -e '#define ClingWorkAroundNoPrivateClassIO' -e '#define ClingWorkAroundUnloadingVTABLES' -e 'gSystem->SetBuildDir("/build/jenkins/workspace/lcg_nightly_pipeline/build/projects/ROOT-HEAD/src/ROOT-HEAD-build/roottest/cling/parsing",true)' -e 'gSystem->AddDynamicPath("/build/jenkins/workspace/lcg_nightly_pipeline/build/projects/ROOT-HEAD/src/ROOT-HEAD-build/roottest/cling/parsing")' -e 'gROOT->SetMacroPath("/build/jenkins/workspace/lcg_nightly_pipeline/build/projects/ROOT-HEAD/src/ROOT/HEAD/roottest/cling/parsing")' -e 'gInterpreter->AddIncludePath("-I/build/jenkins/workspace/lcg_nightly_pipeline/build/projects/ROOT-HEAD/src/ROOT-HEAD-build/roottest/cling/parsing")' -e 'gSystem->AddIncludePath("-I/build/jenkins/workspace/lcg_nightly_pipeline/build/projects/ROOT-HEAD/src/ROOT-HEAD-build/roottest/cling/parsing")' -q -l -b /build/jenkins/workspace/lcg_nightly_pipeline/build/projects/ROOT-HEAD/src/ROOT/HEAD/roottest/cling/parsing/semicolon.C("abc)
-- BEGIN TEST OUTPUT --

Processing /build/jenkins/workspace/lcg_nightly_pipeline/build/projects/ROOT-HEAD/src/ROOT/HEAD/roottest/cling/parsing/semicolon.C("abc)...
Error in <TUnixSystem::SplitAclicMode>: Cannot parse argument in /build/jenkins/workspace/lcg_nightly_pipeline/build/projects/ROOT-HEAD/src/ROOT/HEAD/roottest/cling/parsing/semicolon.C("abc)
Error in <TApplication::ExecuteFile>: macro /build/jenkins/workspace/lcg_nightly_pipeline/build/projects/ROOT-HEAD/src/ROOT/HEAD/roottest/cling/parsing/semicolon.C("abc) not found in path /build/jenkins/workspace/lcg_nightly_pipeline/build/projects/ROOT-HEAD/src/ROOT/HEAD/roottest/cling/parsing

-- END TEST OUTPUT --
CMake Error at /build/jenkins/workspace/lcg_nightly_pipeline/build/projects/ROOT-HEAD/src/ROOT-HEAD-build/RootTestDriver.cmake:186 (message):
  got exit code 1 but expected 0
```

I did not manage at first to run the semicolon test, since it just didn't exist in my local build. After I changed the name to semicollon, I could run the test (with its new name). I did not see the test mentioned in the one CI log I looked at, so I am wondering if anyone else is able to run this test?

(I have no clue how we manage run this test in our nightlies, maybe depends on the cmake version).

The semicolon in the MACROARG is also not working to be properly passed forward. 

### Reproducer

ctest -R semicolon

### ROOT version

master

### Installation method

from source

### Operating system

linux

### Additional context

```
cmake --version
cmake version 3.26.5
```

--- a/roottest/cling/parsing/CMakeLists.txt
+++ b/roottest/cling/parsing/CMakeLists.txt
@@ -30,11 +30,13 @@ ROOTTEST_ADD_TEST(runTTestClass
                   DEPENDS ${GENERATE_DICTIONARY_TEST}
                   LABELS roottest regression cling)
 
-ROOTTEST_ADD_TEST(semicolon
-                  MACRO semicolon.C
-                  MACROARG \"abc;xyz\"
-                  OUTREF semicolon.ref
-                  LABELS roottest regression cling)
+if(NOT MSVC) # See #20738
+    ROOTTEST_ADD_TEST(thesemicolon
+                    COPY_TO_BUILDDIR semicolon.C ./thesemicolon.sh
+                    COMMAND ./thesemicolon.sh
+                    OUTREF semicolon.ref
+                    LABELS roottest regression cling)
+endif()
 
 ROOTTEST_ADD_TEST(assertManipulators
                   MACRO assertManipulators.C

--- a/roottest/cling/parsing/thesemicolon.sh
+++ b/roottest/cling/parsing/thesemicolon.sh
@@ -1,0 +1,3 @@
+#! /usr/bin/env sh
+
+root.exe -b -l -q semicolon.C\(\"abc\ \;\ xyz\"\)

